browser-switch/src/main/java/com/braintreepayments/browserswitch/BrowserSwitchClient.java

```diff
@@ -0,0 +1,332 @@
+package com.braintreepayments.browserswitch;
+
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+
+import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentActivity;
+
+@SuppressWarnings("WeakerAccess")
+public class BrowserSwitchClient {
+
+    private BrowserSwitchConfig config;
+    private ActivityFinder activityFinder;
+    private BrowserSwitchPersistentStore persistentStore;
+    private String returnUrlScheme;
+
+    public static BrowserSwitchClient newInstance(String returnUrlScheme) {
+        return new BrowserSwitchClient(
+            BrowserSwitchConfig.newInstance(), ActivityFinder.newInstance(),
+            BrowserSwitchPersistentStore.getInstance(), returnUrlScheme
+        );
+    }
+
+    @VisibleForTesting
+    static BrowserSwitchClient newInstance(
+        BrowserSwitchConfig config, ActivityFinder activityFinder,
+        BrowserSwitchPersistentStore persistentStore, String returnUrlScheme
+    ) {
```

Comment:
Having the `{` on a newline isn't consistent with rest of repo

Reply:
Yeah this is something to address in our style guidelines. I'm up for placing the `) {` as a trailing charset, the main deal here was to try to meet the 100 column limit.